fluss-common/src/test/java/org/apache/fluss/row/columnar/ColumnarMapTest.java

```diff
@@ -0,0 +1,338 @@
+        ColumnarMap map2 = new ColumnarMap(keyArray2, valueArray2);
+
+        assertThat(map1.equals(map2)).isFalse();
+        assertThat(map1.equals(null)).isFalse();
+        assertThat(map1.equals("string")).isFalse();
+    }
+
+    @Test
+    public void testHashCode() {
+        Object[] keys = {BinaryString.fromString("key1"), BinaryString.fromString("key2")};
+        Object[] values = {100, 200};
+
+        InternalArray keyArray = new GenericArray(keys);
+        InternalArray valueArray = new GenericArray(values);
+
+        ColumnarMap map1 = new ColumnarMap(keyArray, valueArray);
+        ColumnarMap map2 = new ColumnarMap(keyArray, valueArray);
+
+        assertThat(map1.hashCode()).isEqualTo(map2.hashCode());
+    }
+
+    @Test
+    public void testToString() {
+        Object[] keys = {BinaryString.fromString("k1")};
+        Object[] values = {42};
+
+        InternalArray keyArray = new GenericArray(keys);
+        InternalArray valueArray = new GenericArray(values);
+
+        ColumnarMap map = new ColumnarMap(keyArray, valueArray);
+
+        String result = map.toString();
+        assertThat(result).contains("ColumnarMap");
+        assertThat(result).contains("keyArray");
+        assertThat(result).contains("valueArray");
+    }
+
+    @Test
+    public void testMapWithNullValues() {
+        Object[] keys = {
+            BinaryString.fromString("k1"),
+            BinaryString.fromString("k2"),
+            BinaryString.fromString("k3")
+        };
+        Object[] values = {1, null, 3};
+
+        InternalArray keyArray = new GenericArray(keys);
+        InternalArray valueArray = new GenericArray(values);
+
+        ColumnarMap map = new ColumnarMap(keyArray, valueArray);
+
+        assertThat(map.size()).isEqualTo(3);
+        assertThat(map.valueArray().isNullAt(0)).isFalse();
+        assertThat(map.valueArray().isNullAt(1)).isTrue();
+        assertThat(map.valueArray().isNullAt(2)).isFalse();
+    }
+
+    @Test
+    public void testMapWithDifferentDataTypes() {
+        long[] keys = {1L, 2L, 3L};
+        double[] values = {1.1, 2.2, 3.3};
+
+        TestLongColumnVector keyColumnVector = new TestLongColumnVector(keys);
+        TestDoubleColumnVector valueColumnVector = new TestDoubleColumnVector(values);
+
+        ColumnarMap map = new ColumnarMap(keyColumnVector, valueColumnVector, 0, 3);
+
+        assertThat(map.size()).isEqualTo(3);
+        assertThat(map.keyArray().getLong(0)).isEqualTo(1L);
+        assertThat(map.valueArray().getDouble(0)).isEqualTo(1.1);
+        assertThat(map.keyArray().getLong(2)).isEqualTo(3L);
+        assertThat(map.valueArray().getDouble(2)).isEqualTo(3.3);
+    }
+
+    private static class TestIntColumnVector implements IntColumnVector {
+        private final int[] values;
+        private final boolean[] nulls;
+
+        TestIntColumnVector(int[] values) {
+            this.values = values;
+            this.nulls = new boolean[values.length];
```

Comment:
as this is a primitive array, all elements should be not-null, rather than null.